src/app/public/theming/theme-if.directive.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  Directive,
+  Input,
+  OnDestroy,
+  TemplateRef,
+  ViewContainerRef
+} from '@angular/core';
+
+import {
+  Subject
+} from 'rxjs';
+
+import {
+  takeUntil
+} from 'rxjs/operators';
+
+import {
+  SkyThemeSettings
+} from './theme-settings';
+
+import {
+  SkyThemeService
+} from './theme.service';
+
+/**
+ * Component that works like `ngIf` to show markup for matching theme.
+ *
+ * If the directive is within a `skyTheme` directive, it uses settings from that directive.
+ */
+@Directive({
+  selector: '[skyThemeIf]'
+})
+export class SkyThemeIfDirective implements OnDestroy {
+  private context: string;
+  private currentTheme: SkyThemeSettings | undefined;
+  private ngUnsubscribe = new Subject();
+  private hasView = false;
+
+  constructor(
+    private themeSvc: SkyThemeService,
+    private templateRef: TemplateRef<any>,
+    private viewContainer: ViewContainerRef
+  ) {
+    this.themeSvc.settingsChange
+      .pipe(takeUntil(this.ngUnsubscribe))
+      .subscribe(settingsChange => {
+        this.themeSettings = settingsChange.currentSettings;
+      });
+  }
+
+  public ngOnDestroy(): void {
+    this.ngUnsubscribe.next();
+    this.ngUnsubscribe.complete();
+  }
+
+  /**
+   * A string that should match the name of a theme, `'default'` or `'modern'`.
+   *
+   * @param value
+   */
+  @Input()
+  set skyThemeIf(value: 'default' | 'modern') {
```

Comment:
Same comment as above - add `public` to this property and make sure all properties are above the constructor.